### PR TITLE
P4-2962 using chromedriver to resolve issue with JavaScript not working. Price add and update pages now following correct path.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
**Changes:**

The changes in this PR now mean we are using chromedriver to drive the integration tests.  This resolves the issue around JavaScript not working (with HtmlUnit).

Both the add and update price screens now follow the correct flows through the application.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -56,6 +56,10 @@ dependencies {
   testImplementation("com.squareup.okhttp3:okhttp:4.9.1")
   testImplementation("org.testcontainers:postgresql:1.15.3")
 
+  testImplementation("org.seleniumhq.selenium:selenium-support:3.141.59")
+  testImplementation("org.seleniumhq.selenium:selenium-chrome-driver:3.141.59")
+  testImplementation("org.seleniumhq.selenium:selenium-firefox-driver:3.141.59")
```
Comment (matt-crowson-moj) [inline build.gradle.kts:61]: I'll remove this (Firefox driver) in a separate PR.

**Review by thomasleese — COMMENTED** [review]
The change makes sense to me, but I’ll leave it to someone else to approve!

I was just wandering if it was worth splitting the commit up into smaller ones? I’m thinking perhaps the introduction of chromedriver in one and then the utilisation of it in another?

**Comment by matt-crowson-moj** [comment]
> The change makes sense to me, but I’ll leave it to someone else to approve!
> 
> I was just wandering if it was worth splitting the commit up into smaller ones? I’m thinking perhaps the introduction of chromedriver in one and then the utilisation of it in another?

Thanks Tom but in short no.  The changes are dependent on the introduction of the Chromedriver.  The CI pipeline needs the change at the same time or the build will fail and can't be merged.

**Comment by thomasleese** [comment]
> The changes are dependent on the introduction of the Chromedriver. The CI pipeline needs the change at the same time or the build will fail and can't be merged.

I was imagining that it's okay if the build _would_ fail on an individual commit in the PR, as long as the final commit and merge commit into main build correctly. I tend to prefer lots of smaller commits to make it easier to do `git blame` or `git bisect` in the future or something like that. Either way, I don't really mind too much.

I have thought of a second question though, probably not for this PR, but to be noted down somewhere, is it worth us switching all the tests over in this repo to run using `chromedriver`? It might avoid some potential confusion where it's unclear how the tests are being run, and keeps things nice and consistent. I imagine a downside of this is that the tests take longer to run.

Further down the line, it might also allow us to do something clever in the future if we know the tests run under `chromedriver` we could try running them under `geckodriver` as well to be aware of any potential browser differences?

**Comment by matt-crowson-moj** [comment]
> > The changes are dependent on the introduction of the Chromedriver. The CI pipeline needs the change at the same time or the build will fail and can't be merged.
> 
> I was imagining that it's okay if the build _would_ fail on an individual commit in the PR, as long as the final commit and merge commit into main build correctly. I tend to prefer lots of smaller commits to make it easier to do `git blame` or `git bisect` in the future or something like that. Either way, I don't really mind too much.
> 
> I have thought of a second question though, probably not for this PR, but to be noted down somewhere, is it worth us switching all the tests over in this repo to run using `chromedriver`? It might avoid some potential confusion where it's unclear how the tests are being run, and keeps things nice and consistent. I imagine a downside of this is that the tests take longer to run.
> 
> Further down the line, it might also allow us to do something clever in the future if we know the tests run under `chromedriver` we could try running them under `geckodriver` as well to be aware of any potential browser differences?

All tests bar one are now using Chromedriver.  There is one test using a different driver (for now) due to testing a file download.  This is an acceptable trade off for now.  The thing to be aware of here this has only recently been added (integration testing that is).  So with regards to different browsers sure, certainly that is something to consider going forwards. The main driver here however was getting a framework in place to enable us to do that in future.

**Review by Thource — APPROVED** [review]

**Comment by thomasleese** [comment]
> All tests bar one are now using Chromedriver. There is one test using a different driver (for now) due to testing a file download. This is an acceptable trade off for now. The thing to be aware of here this has only recently been added (integration testing that is). So with regards to different browsers sure, certainly that is something to consider going forwards. The main driver here however was getting a framework in place to enable us to do that in future.

Oh sorry, I'd misunderstood and thought this PR was about adding Chromedriver for one of the tests, but I see it's actually about adding Chromedriver for all the tests except the one which doesn't work with it!

**Pull request merged**